### ISSUE TITLE
Crop/Upload example filename = undefined

### ISSUE THREAD

**Issue by iamelru** [danialfarid/ng-file-upload#1359]
Hello! Thanx for your work!

http://jsfiddle.net/xxo3sk41/1/
even in fiddle it does not send a filename:

------WebKitFormBoundarycAukxnCKLt0epLLk
Content-Disposition: form-data; name="file"; filename="undefined"
Content-Type: image/png

**Comment by danialfarid** [comment]
Use `Upload.rename()` before uploading the file to set the correct original name for it.

**Comment by iamelru** [comment]
Thank you

i simply added picFile 
`ng-click="upload(croppedDataUrl, picFile)"`
and then used `picFile.name` in
```
data: {
                file: Upload.dataUrltoBlob(dataUrl, picfile.name)
            }
```

**Comment by danialfarid** [comment]
Fixed at 12.0.3

**Comment by djindjic** [comment]
I have really researched all issues and docs but can't realized how I can change name="file" to name="newName"?
